### PR TITLE
Move mock API classes to tests

### PR DESCRIPTION
**Pull request by RishiDiwanTT**
## Description
Moved mock api classes into the tests/api folder.
Removed some unused classes from the testing.py files.
Deleted the "DatabaseTest" class, left a single method that is still in use.
Moved pytest to a dev dependency in poetry.
An autouse db session fixture was not in use, deleted that too.
Both core.testing and api.testing have been moved to the tests module.
A new MockCirculationManager should be used when testing the CirculationManager methods.

## Motivation and Context
Currently some test dependancies have to be installed in our production instances, because the mocks for our test code, live beside the production code. This bloats the attack surface of our application, make the production server spend more time importing libraries and generally makes it harder to refactor these mocks, as they could cause issues in production code.

This PR has changes for 2 tickets
- Remove pytest from production code [Notion](https://www.notion.so/lyrasis/Refactor-circulation-tests-so-mocks-are-not-imported-with-production-code-b2602b1b8c8d475995f1fd119eea37f7?pvs=4)
- Cleanup old code and use fixtures [Notion](https://www.notion.so/lyrasis/Migrate-remaining-CM-test-code-to-fixtures-and-clean-up-old-style-test-classes-cc7a3d222565404193c08c3740459358?pvs=4)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by jonathangreen — COMMENTED** [review]
Couple little comments. Mostly hoping we can move a bit more code over.

File: tests/fixtures/db.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+from core.model import Collection, ExternalIntegration, Library, get_one_or_create
+
+
+class DatabaseTest:
+    """DEPRECATED DO NOT USE
+    This only exists because its one method is still in use"""
+
+    @classmethod
+    def make_default_library(cls, _db):
```
[the remark this answers is not in the document]
Comment (RishiDiwanTT) [reply tests/fixtures/db.py:11]: I tried doing this, the problem is the [mock enki](https://github.com/ThePalaceProject/circulation/pull/1049/files#diff-0dfc9aa0df7312f58f50f733448cd721311b7aa6f81312f4eff77cc121124950R8) class.
It is used from a monitor class(api_class=MockEnki) in the testing code so cannot be passed a fixture from the production monitor class code.
I can remove the DatabaseTest class and keep this as a standalone function.

**Comment by RishiDiwanTT** [comment]
@dbernstein I noticed the [test_s3_analytics](https://github.com/ThePalaceProject/circulation/blob/main/tests/test_s3_analytics_provider.py) file which is in neither the core and api folder, and hence will not be run ever.
I have not changed this file due to that reason.

File: api/testing.py [hunk 2]
```diff
@@ -1,136 +1,9 @@
...
-    expired["id"] = "expired"
-    expired["start"] = a_week_ago
-    expired["finish"] = yesterday
-
-    # This announcement should be displayed starting tomorrow.
-    forthcoming = dict(active)
-    forthcoming["id"] = "forthcoming"
-    forthcoming["start"] = tomorrow
-    forthcoming["finish"] = in_a_week
+from core.model import DataSource, Hold, Loan
```
[the remark this answers is not in the document]
Comment (RishiDiwanTT) [reply api/testing.py:8]: This took some finagling of the CirculationManager to remove, but everything works now :+1:

File: api/controller.py [hunk 3]
```diff
@@ -356,29 +349,18 @@ def log_lanes(self, lanelist=None, level=0):
...
     def setup_search(self):
         """Set up a search client."""
-        if self.testing:
-            return MockExternalSearchIndex()
-        else:
-            search = ExternalSearchIndex(self._db)
-            if not search:
-                self.log.warn("No external search server configured.")
-                return None
-            return search
```
Comment (jonathangreen) [inline api/controller.py:366]: 🎉

File: api/controller.py [hunk 4]
```diff
@@ -356,29 +349,18 @@ def log_lanes(self, lanelist=None, level=0):
...
+            return None
+        return search
 
     def setup_circulation(self, library, analytics):
         """Set up the Circulation object."""
-        if self.testing:
-            cls = MockCirculationAPI
-        else:
-            cls = CirculationAPI
-        return cls(self._db, library, analytics)
```
Comment (jonathangreen) [inline api/controller.py:374]: 🚀

File: api/controller.py [hunk 5]
```diff
@@ -356,29 +349,18 @@ def log_lanes(self, lanelist=None, level=0):
...
+        return CirculationAPI(self._db, library, analytics)
 
     def setup_shared_collection(self):
-        if self.testing:
-            cls = MockSharedCollectionAPI
-        else:
-            cls = SharedCollectionAPI
-        return cls(self._db)
```
Comment (jonathangreen) [inline api/controller.py:381]: 🤩

**Review by jonathangreen — APPROVED** [review]
!()[https://media.giphy.com/media/2K4V9yBmFjjKo/giphy.gif]

**Comment by dbernstein** [comment]
@RishiDiwanTT :  yes - please move it (and update it to work with your changes).   Sorry for the delayed response. And good catch!

**Pull request merged**